pandas/core/frame.py

```diff
@@ -703,14 +703,15 @@ def itertuples(self, index=True, name="Pandas"):
 
         # Python 3 supports at most 255 arguments to constructor, and
         # things get slow with this many fields in Python 2
-        if len(self.columns) + index < 256:
+        if name is not None and len(self.columns) + index < 256:
             # `rename` is unsupported in Python 2.6
             try:
                 itertuple = collections.namedtuple(
                     name, fields+list(self.columns), rename=True)
-                return (itertuple(*row) for row in zip(*arrays))
-            except:
+            except Exception:
                 pass
+            else:
```

Comment:
Sorry for correcting you, but the code does exactly what you say is to be achieved.
> The try ... except statement has an optional else clause, which, when present,
> must follow all except clauses. It is useful for code that must be executed
> if the try clause **does not raise** an exception.